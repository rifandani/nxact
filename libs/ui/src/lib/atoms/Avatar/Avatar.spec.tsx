import {
  Avatar,
  AvatarFallback,
  AvatarImage,
} from 'libs/ui/src/lib/atoms/Avatar/Avatar';
import renderer from 'react-test-renderer';

describe('Avatar component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Avatar>
          <AvatarImage
            src="https://github.com/rifandani.png"
            alt="@rifandani"
          />
          <AvatarFallback>RR</AvatarFallback>
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render fallback if no image component provided', () => {
    const tree = renderer
      .create(
        <Avatar>
          <Avatar>RR</Avatar>
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
