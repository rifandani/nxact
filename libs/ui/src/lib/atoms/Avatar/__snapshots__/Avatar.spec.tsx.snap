// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render correctly 1`] = `
<span
  className="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
>
  <span
    className="flex h-full w-full items-center justify-center rounded-full bg-slate-100 dark:bg-slate-700"
  >
    RR
  </span>
</span>
`;

exports[`Avatar component should render fallback if no image component provided 1`] = `
<span
  className="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
>
  <span
    className="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
  >
    RR
  </span>
</span>
`;
